### PR TITLE
Fixes #1795 New 'enter' in comment feature indents incorrectly

### LINKED ISSUE #1795: New 'enter' in comment feature indents incorrectly
Before:
```
names = ['john', 'jane']

try:
    for name in names:
        # hello this is a comment, press enter here to see what happens and then here again
        pass

except:
    pass
```

After first enter:
```
names = ['john', 'jane']

try:
    for name in names:
        # hello this is a comment, press enter here 
                #to see what happens and then here again
        pass

except:
    pass
```

After second enter:
```
names = ['john', 'jane']

try:
    for name in names:
        # hello this is a comment, press enter here 
                #to see what happens and then here
                               # again
        pass

except:
    pass
```

### PR DESCRIPTION
**Pull request by zooba**
Fixes #1795 New 'enter' in comment feature indents incorrectly
Fixes indentation
Fixes issues with mock test infrastructure
Prevents deadlock on exit due to error task providers (attempt #3)

### CODE REVIEW

File: Common/Tests/MockVsTests/MockVsServiceProvider.cs [hunk 1]
```diff
@@ -57,38 +52,36 @@ public class MockVsServiceProvider : SVsServiceProvider, IOleServiceProvider, IS
...
 
-            Console.WriteLine("Unknown service: " + serviceType.FullName);
+            var msg = string.Format(
+                "Unknown Service {0} ({1:B})",
+                Type.GetTypeFromCLSID(serviceType, false)?.FullName ?? "(unknown)",
+                serviceType
+            );
+
+            Console.WriteLine(msg);
+            //throw new NotImplementedException(msg);
```
Comment (huguesv) [inline Common/Tests/MockVsTests/MockVsServiceProvider.cs]: remove

File: Python/Product/VSInterpreters/CPythonInterpreterFactoryProvider.cs [hunk 2]
```diff
@@ -32,7 +32,7 @@ class CPythonInterpreterFactoryProvider : IPythonInterpreterFactoryProvider {
         private readonly Dictionary<string, PythonInterpreterInformation> _factories = new Dictionary<string, PythonInterpreterInformation>();
         const string PythonPath = "Software\\Python";
         internal const string FactoryProviderName = "Global";
-        private bool _watchRegistry;
+        internal bool _watchRegistry;
```
Comment (huguesv) [inline Python/Product/VSInterpreters/CPythonInterpreterFactoryProvider.cs]: is the change to internal needed? the PR doesn't seem to add any use of the member

**Comment by huguesv** [comment]
👍

File: Python/Product/VSInterpreters/CPythonInterpreterFactoryProvider.cs (hunk 2 above)
Comment (zooba) [inline Python/Product/VSInterpreters/CPythonInterpreterFactoryProvider.cs]: Ah, you're right. I was going to try and change it from a test but it turned out to be unnecessary. I'll revert

**Pull request merged**